entity.go

```diff
@@ -56,6 +56,16 @@ func (e BasicEntity) ID() uint64 {
 	return e.id
 }
 
+//By having this, All Entities, with this component, now have a GetBasicEntity Method
```

Comment:
I like your explanation but if it can follow [godoc style documentation](https://blog.golang.org/godoc-documenting-go-code), that'd be even better. Thanks.